zeppelin-interpreter/src/test/java/org/apache/zeppelin/cluster/event/ClusterMessageSerDeserTest.java

```diff
@@ -0,0 +1,156 @@
+        assertNull(deserialized.getMsgId());
+    }
+
+    @Test
+    void fromJson_withNothing() {
+        // When only the empty JSON object is provided,
+        ClusterMessage deserialized = ClusterMessage.deserializeMessage("{}");
+
+        // Then, all fields should be null or empty
+        assertNull(deserialized.clusterEvent);
+        assertNull(deserialized.getData());
+        assertNull(deserialized.getMsgId());
+    }
+
+    @Test
+    void fromJson_withNullFields() {
+        String jsonWithNullFields = a2q("{" +
+                "'clusterEvent':null," +
+                "'data':null," +
+                "'msgId':null" +
+                "}");
+
+        ClusterMessage deserialized = ClusterMessage.deserializeMessage(jsonWithNullFields);
+        assertNull(deserialized.clusterEvent);
+        assertNull(deserialized.getData());
+        assertNull(deserialized.getMsgId());
+    }
+
+    @Test
+    void toJson_withEmptyData() {
+        ClusterMessage original = new ClusterMessage(ClusterEvent.CLEAR_PERMISSION);
+        original.setMsgId("");
+        original.getData().clear();
+
+        String serializedJSON = ClusterMessage.serializeMessage(original);
+        assertEquals(
+                a2q("{\n" +
+                        "  'clusterEvent': 'CLEAR_PERMISSION',\n" +
+                        "  'data': {},\n" +
+                        "  'msgId': ''\n" +
+                        "}"),
+                serializedJSON);
+
+        ClusterMessage deserialized = ClusterMessage.deserializeMessage(serializedJSON);
+        assertEquals(original.clusterEvent, deserialized.clusterEvent);
+        assertEquals(original.getData(), deserialized.getData());
+        assertEquals(original.getMsgId(), deserialized.getMsgId());
+    }
+
+    @Test
+    void fromJson_withLargeDataSet() {
+        Map<String, String> largeData = new HashMap<>();
+        for (int i = 0; i < 1000; i++) {
+            largeData.put("key" + i, "value" + i);
+        }
+
+        ClusterMessage original = new ClusterMessage(ClusterEvent.SET_READERS_PERMISSIONS);
+        original.setMsgId(UUID.randomUUID().toString());
+        original.put(largeData);
+
+        String serializedJSON = ClusterMessage.serializeMessage(original);
+        ClusterMessage deserialized = ClusterMessage.deserializeMessage(serializedJSON);
+
+        assertEquals(original.clusterEvent, deserialized.clusterEvent);
+        assertEquals(original.getData(), deserialized.getData());
+        assertEquals(original.getMsgId(), deserialized.getMsgId());
+    }
+
+    private Map<String, String> mapOf(
+            String key, String value,
+            String key2, String value2
+    ) {
+        Map<String, String> map = new HashMap<>();
+        map.put(key, value);
+        map.put(key2, value2);
+        return map;
+    }
+
+    private static String a2q(String json) {
+        return json.replace("'", "\"");
+    }
```

Comment:
Is a2q really necessary? If not, I think using a pure JSON format without a2q could make the test clearer. Keeping the test focused on the target, which in this case is JSON serialization/deserialization, could help keep the test simple. 
Also, in this PR HeliumRegistrySerializerTest does not use a2q.

Reply:
> Is a2q really necessary? If not, I think using a pure JSON format without a2q could make the test clearer. Keeping the test focused on the target, which in this case is JSON serialization/deserialization, could help keep the test simple. 

Hmmm..  You are right method `a2q()` sure is not necessary. But from a different point of view, `a2q()` is just a little helper method to make it easier to read the json input so we can focus more on JSON serialization/deserialization. Some people do use helper methods to actually focus on the SUT in this case the JSON serialization/deserialization. 

What I would like to see from you is some guidelines or references on how helper methods should be written so tests are more on focus.

> Also, in this PR HeliumRegistrySerializerTest does not use a2q.

Agreed 👏🏽👏🏽 I will use `a2q` in all tests introduced in this PR

Reply:
Sure I will remove a2q methods.
I disagree with you and a2q methods don't hinder us from focusing on the SUT.
I just don't have much preference over `a2q` here.